### PR TITLE
Update Redumper to build 311

### PR DESCRIPTION
**Pull request by Deterous**
Commands added: Rings, Dumpnew, Refinenew (Dumpnew and Refinenew are temporary, as they are WIP)
Flags added: Version, DumpWriteOffset, LegacySubs, DisableCDText
Flags renamed: PlextorLeadinSkip -> PlextorSkipLeadin

### CODE REVIEW

File: MPF.Core/Modules/Redumper/Parameters.cs [hunk 1]
```diff
@@ -772,6 +781,13 @@ public override string GenerateParameters()
                     parameters.Add($"{FlagStrings.Skip}={SkipValue}");
             }
 
+            // Dump Write Offset
+            if (this[FlagStrings.DumpWriteOffset] == true)
+            {
+                if (DumpWriteOffsetValue != null && DumpWriteOffsetValue > 0)
```
Comment (mnadareski) [inline MPF.Core/Modules/Redumper/Parameters.cs:787]: Does the write offset have to be positive?

File: MPF.Core/Modules/Redumper/Parameters.cs [hunk 2]
```diff
@@ -1306,14 +1340,25 @@ protected override bool ValidateAndSetParameters(string? parameters)
                 if (!string.IsNullOrEmpty(stringValue))
                     SkipValue = stringValue;
 
-                // Skip
+                // Dump Write Offset
+                intValue = ProcessInt32Parameter(parts, FlagStrings.DumpWriteOffset, ref i);
+                if (!string.IsNullOrEmpty(stringValue))
```
Comment (mnadareski) [inline MPF.Core/Modules/Redumper/Parameters.cs:1345]: This and the following are invalid checks. Both should use the same checks that LBA Start and End are using.

File: MPF.Core/Modules/Redumper/Parameters.cs (hunk 1 above)
Comment (Deterous) [reply MPF.Core/Modules/Redumper/Parameters.cs:787]: No i'm just a dumb copypaste machine

File: MPF.Core/Modules/Redumper/Parameters.cs (hunk 2 above)
Comment (Deterous) [reply MPF.Core/Modules/Redumper/Parameters.cs:1345]: Why `!= Int32.MinValue` ?

Comment (mnadareski) [reply MPF.Core/Modules/Redumper/Parameters.cs:1345]: https://github.com/SabreTools/MPF/blob/65ad629ee0c90f77c925fc854b6bc653d4399943/MPF.Core/Modules/BaseParameters.cs#L755C17-L755C17

Comment (Deterous) [reply MPF.Core/Modules/Redumper/Parameters.cs:1345]: Oh I see

**Review by mnadareski — APPROVED** [review]

**Pull request merged**